test/state.cpp

```diff
@@ -154,6 +160,64 @@ TEST_CASE_FIXTURE(StateTest, "Full Size Group")
   }
 }
 
+TEST_CASE_FIXTURE(StateTest, "Roster Updates")
+{
+  // Initialize the creator's state
+  states.emplace_back(
+    group_id, suite, init_privs[0], identity_privs[0], key_packages[0]);
+
+  // Each participant invites the next
+  for (size_t i = 1; i < group_size; i += 1) {
+    auto sender = i - 1;
+
+    auto add = states[sender].add(key_packages[i]);
+    states[sender].handle(add);
+
+    auto [commit, welcome, new_state] = states[sender].commit(fresh_secret());
+    for (size_t j = 0; j < states.size(); j += 1) {
+      if (j == sender) {
+        states[j] = new_state;
+      } else {
+        states[j].handle(add);
+        states[j] = states[j].handle(commit).value();
+      }
+    }
+
+    states.emplace_back(
+      init_privs[i], identity_privs[i], key_packages[i], welcome);
+
+    // Check that everyone ended up in the same place
+    for (const auto& state : states) {
+      REQUIRE(state == states[0]);
+    }
+
+    verify_group_functionality(states);
+  }
+
+  // remove member at position 1
+  auto remove_1 = states[0].remove(RosterIndex{ 1 });
+  states[0].handle(remove_1);
+
+  // remove member at position 2
+  auto remove_2 = states[0].remove(RosterIndex{ 2 });
+  states[0].handle(remove_2);
+
+  // commit to new state
+  auto [commit, welcome, new_state] = states[0].commit(fresh_secret());
+  silence_unused(welcome);
+
+  auto roster_ref = new_state.roster();
+  REQUIRE(3 == roster_ref.size());
```

Comment:
You can do better than this by checking the actual contents of the roster.
```
  auto expected_roster = std::vector<Credential>{
    key_packages[0].credential,
    key_packages[2].credential,
    key_packages[4].credential,
  };
  REQUIRE(expected_roster = new_state.roster());
```

I think you will then find the bug I mention above :)